kubectl: add port-forward svc/ completion

sroskelley opened the pull request:
**What this PR does / why we need it**:
adds bash completion for 
`kubectl port-forward svc/<tab>`
`kubectl logs <tab>`

**Which issue(s) this PR fixes**:
Fixes #79718

**Special notes for your reviewer**:
Thanks for your work. 

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

/kind feature
/sig cli

m1093782566 committed:
Add API for feature gate ServiceTopology

aramase committed:
Allow multiple node cidr masks in cm

update tests

add comment

amend var name

update comment

add check for empty slice

fix tests

fix mask size in test

review feedback

add ipv4 and ipv6 flag for mask sizes

add to violation exception list

remove import alias

run update-openapi-spec

review feedback

run update-bazel

review feedback

review feedback

imroc committed:
Service Topology implementation

* Implement Service Topology for ipvs and iptables proxier
* Add test files
* API validation

sroskelley committed:
add port-forward svc/ completion

sroskelley committed:
adding better logs tab completion.

sroskelley commented:
/assign @adohe

thockin committed:
Bump debian-base to buster

danwinship committed:
Add mode-detecting iptables wrappers to the debian-iptables image

ddebroy committed:
Disable in-tree plugins migrated to CSI

Signed-off-by: Deep Debroy <ddebroy@docker.com>

claudiubelu committed:
tests: Adds large requests tests

Ensures that requests that require large packets work properly, and that
they are not dropped.

Adds AgnhostPrivate to test/utils/image/manifest. Some tests are trying to pull
the agnhost image from the private registry, meaning that we would need to
always build and push the agnhost image to both e2e and private registry
whenever we bump its version. Decoupling them would mean that we only need
to push the image to the e2e registry.

codenrhoden committed:
Retire mount.Exec for k8s.io/utils/exec

This patch removes mount.Exec entirely and instead uses the common
utility from k8s.io/utils/exec.

The fake exec implementation found in k8s.io/utils/exec differs a bit
than mount.Exec, with the ability to pre-script expected calls to
Command.CombinedOutput(), so tests that previously relied on a callback
mechanism to produce specific output have been updated to use that
mechanism.

RainbowMango committed:
Hide scheduler metrics that have been deprecated in 1.14

RainbowMango committed:
Hide apiserver metrics that have been deprecated in 1.14
Update E2E test accordingly.

RainbowMango committed:
Hide apiserver metrics that have been deprecated in 1.15

liggitt committed:
Add support for --runtime-config=api/beta=false, --feature-gates=AllBeta=false

Allow disabling all beta features and APIs

robscott committed:
Promoting EndpointSlices to beta

mikedanese committed:
Add an expiring cache for the caching token authenticator

And maybe the webhook authorizer cache.

This cache has two primary advantages over the LRU cache used currently:

- Cache hits don't acquire an exclusive lock.
- More importantly, performance doesn't fallover when the access pattern
  scans a key space larger than an arbitrary size (e.g. the LRU
  capacity).

The downside of using an expiring cache here is that it doesn't have a
maximum size so it's suspectible to DoS when the input is user
controlled. This is not the case for successful authentications, and
successful authentications have a natural expiry so it might be a good
fit here.

It has some a few differences compared to:

https://github.com/kubernetes/kubernetes/blob/3d7318f29d9f56810efd3d690811cdea730b5317/staging/src/k8s.io/client-go/tools/cache/expiration_cache.go

- Expiration is not entirely lazy so keys that are never accessed again
  are still released from the cache.
- It does not acquire an exclusive lock on cache hits.
- It supports per entry ttls specified on Set.

The expiring cache (without striping) does somewhere in between the
simple cache and striped cache in the very contrived contention test
where every iteration acquires a write lock:

```
$ benchstat simple.log expiring.log
name      old time/op    new time/op    delta
Cache-12    2.74µs ± 2%    2.02µs ± 3%  -26.37%  (p=0.000 n=9+9)
name      old alloc/op   new alloc/op   delta
Cache-12      182B ± 0%      107B ± 4%  -41.21%  (p=0.000 n=8+9)
name      old allocs/op  new allocs/op  delta
Cache-12      5.00 ± 0%      2.00 ± 0%  -60.00%  (p=0.000 n=10+10)

$ benchstat striped.log expiring.log
name      old time/op    new time/op    delta
Cache-12    1.58µs ± 5%    2.02µs ± 3%  +27.34%  (p=0.000 n=10+9)
name      old alloc/op   new alloc/op   delta
Cache-12      288B ± 0%      107B ± 4%  -62.85%  (p=0.000 n=10+9)
name      old allocs/op  new allocs/op  delta
Cache-12      9.00 ± 0%      2.00 ± 0%  -77.78%  (p=0.000 n=10+10)

$ benchstat simple.log striped.log expiring.log
name \ time/op    simple.log   striped.log  expiring.log
Cache-12          2.74µs ± 2%  1.58µs ± 5%   2.02µs ± 3%
name \ alloc/op   simple.log   striped.log  expiring.log
Cache-12            182B ± 0%    288B ± 0%     107B ± 4%
name \ allocs/op  simple.log   striped.log  expiring.log
Cache-12            5.00 ± 0%    9.00 ± 0%     2.00 ± 0%
```

I also naively replacemed the LRU cache with the expiring cache in the
more realisitc CachedTokenAuthenticator benchmarks:

https://gist.github.com/mikedanese/41192b6eb62106c0758a4f4885bdad53

For token counts that fit in the LRU, expiring cache does better because
it does not require acquiring an exclusive lock for cache hits.

For token counts that exceed the size of the LRU, the LRU has a massive
performance drop off. The LRU cache is around 5x slower (with lookups
taking 1 milisecond and throttled to max 40 lookups in flight).

```
$ benchstat before.log after.log
name                                                  old time/op    new time/op    delta
CachedTokenAuthenticator/tokens=100_threads=256-12      3.60µs ±22%    1.08µs ± 4%  -69.91%  (p=0.000 n=10+10)
CachedTokenAuthenticator/tokens=500_threads=256-12      3.94µs ±19%    1.20µs ± 3%  -69.57%  (p=0.000 n=10+10)
CachedTokenAuthenticator/tokens=2500_threads=256-12     3.07µs ± 6%    1.17µs ± 1%  -61.87%  (p=0.000 n=9+10)
CachedTokenAuthenticator/tokens=12500_threads=256-12    3.16µs ±17%    1.38µs ± 1%  -56.23%  (p=0.000 n=10+10)
CachedTokenAuthenticator/tokens=62500_threads=256-12    15.0µs ± 1%     2.9µs ± 3%  -80.71%  (p=0.000 n=10+10)

name                                                  old alloc/op   new alloc/op   delta
CachedTokenAuthenticator/tokens=100_threads=256-12        337B ± 1%      300B ± 0%  -11.06%  (p=0.000 n=10+8)
CachedTokenAuthenticator/tokens=500_threads=256-12        307B ± 1%      304B ± 0%   -0.96%  (p=0.000 n=9+10)
CachedTokenAuthenticator/tokens=2500_threads=256-12       337B ± 1%      304B ± 0%   -9.79%  (p=0.000 n=10+10)
CachedTokenAuthenticator/tokens=12500_threads=256-12      343B ± 1%      276B ± 0%  -19.58%  (p=0.000 n=10+10)
CachedTokenAuthenticator/tokens=62500_threads=256-12      493B ± 0%      334B ± 0%  -32.12%  (p=0.000 n=10+10)

name                                                  old allocs/op  new allocs/op  delta
CachedTokenAuthenticator/tokens=100_threads=256-12        13.0 ± 0%      11.0 ± 0%  -15.38%  (p=0.000 n=10+10)
CachedTokenAuthenticator/tokens=500_threads=256-12        12.0 ± 0%      11.0 ± 0%   -8.33%  (p=0.000 n=10+10)
CachedTokenAuthenticator/tokens=2500_threads=256-12       13.0 ± 0%      11.0 ± 0%  -15.38%  (p=0.000 n=10+10)
CachedTokenAuthenticator/tokens=12500_threads=256-12      13.0 ± 0%      10.0 ± 0%  -23.08%  (p=0.000 n=9+10)
CachedTokenAuthenticator/tokens=62500_threads=256-12      17.0 ± 0%      12.0 ± 0%  -29.41%  (p=0.000 n=10+10)
```

Benchmarked with changes in #84423

Bugs: #83259 #83375

danielqsj committed:
refactor apparmor utils in e2e

mkimuram committed:
Change publish path for CSI block volume per pod

This change is to allow CSI driver to publish the same volume for multipe pods on the same node.

mkimuram committed:
Separate staging/publish and unstaging/unpublish logics for block

mkimuram committed:
Change globalMapPath to bind mount from symlink

This change is needed to make descriptor lock per pod, in the next commit.
If losetup is called for symlink, path in the output for losetup is resolved,
as a result, we can't distinguish which path the lock is taken.

mkimuram committed:
Make descriptor lock per pod and release it per pod

This change is needed to avoid unmapVolumeFunc for one pod blocked,
when the lock for the same volume is taken for another pod.

mkimuram committed:
Move MapBlockVolume call to operation_generator and add UnmapBlockVolume

hwdef committed:
pkg/client: fix staticcheck warning

danwinship committed:
Update debian-iptables image digests for v12.0.0 build

danwinship committed:
Update debian-base image digests for v2.0.0 build

mkimuram committed:
Refactor BlockVolumeMapper and BlockVolumeUnmapper interface

- Rename MapDevice to MapPodDevice in BlockVolumeMapper
- Add UnmapPodDevice in BlockVolumeUnmapper (This will be used by csi driver later)
- Add CustomBlockVolumeMapper and CustomBlockVolumeUnmapper interface
- Move SetUpDevice and MapPodDevice to CustomBlockVolumeMapper
- Move TearDownDevice and UnmapPodDevice to CustomBlockVolumeUnmapper
- Implement CustomBlockVolumeMapper only in local and csi plugin
- Implement CustomBlockVolumeUnmapper only in fc, iscsi, rbd, and csi plugin
- Change MapPodDevice to return path and SetUpDevice not to return path

mkimuram committed:
Check and return error first in IsSymlinkExist and IsBindMountExist

mkimuram committed:
Improve comments for volume path hanlder and volume.go

mkimuram committed:
Remove klog for output error instead return err with context

imroc committed:
Autogenerated files

sttts committed:
e2e: promote CRD defaulting test to conformance test

Commit:
Update structured-merge-diff to latest version

Commit:
Move RegisterNodeFlags() to e2e_node test

RegisterNodeFlags() is called in e2e_node test package only, so this
moves the function for making e2e core framework small and simple.

davidz627 committed:
Update inline volume translated PV Name to be unique per disk so that staging paths are unique

mikedanese committed:
migrate token cache to cache.Expiring

Commit:
Update dependency vmware/govmomi to v0.20.3

￼
misterikkit committed on Oct 4
govmomi is the vSphere client library used by the vSphere cloud provider
and storage plugin. A bug in the SOAP client prevented storage classes
that use vSphere storage policies (aka SPBM) from working.

This bumps our dependency on vmware/govmomi from v0.20.1 to v0.20.3 to
pick up the fix in vmware/govmomi#1498

Here are all changes in the release:
https://github.com/vmware/govmomi/compare/v0.20.1...v0.20.3

MikeSpreitzer committed:
Enabled discrimination on target namespace

Generalized ResourcePolicyRule in API Priority and Fairness to be able
to discriminate on the target namespace (if any) specified in the
request.

danwinship committed:
Update debian-iptables iptables-wrapper script

The debian-iptables v12.0.0 build didn't work because of another
previously-undiscovered iptables 1.8.x bug. Work around it for now and
bump the version to v12.0.1; we can revert back to the original
version of the script once iptables 1.8.4 is available in
buster-backports.

sttts committed:
apiextension: clarify what atomic means

sttts committed:
apiextensions: clarify validation errors

sttts committed:
apiextensions: validate that list-type set has atomic items

Commit:
add DeleteOptions conversion

aramase committed:
remove disk locks per vm

maintain map with nodename and lock

move lock map to utils

hwdef committed:
pkg/kubeapiserver: fix staticcheck warning

knight42 committed:
test(cr::update): add corresponding tests

Signed-off-by: knight42 <anonymousknight96@gmail.com>

mkimuram committed:
Rename IsBindMountExist to IsDeviceBindMountExist

mkimuram committed:
Remove remaining empty file in unmapBindMountDevice

mkimuram committed:
Fix error messages in operation_generator.go

mkimuram committed:
Change getDeviceMajorMinor to use unix.Stat

mkimuram committed:
Split CustomBlockVolumeMapper and CustomBlockVolumeUnmapper

  - Move SetUpDevice to BlockVolumeStager
  - Move MapPodDevice to BlockVolumePublisher
  - Move TearDownDevice to BlockVolumeUnstager
  - Move UnmapPodDevice to BlockVolumeUnpublisher
  - Implement BlockVolumePublisher only in local and csi plugin
  - Implement BlockVolumeUnstager only in fc, iscsi, rbd, and csi plugin
  - Implement BlockVolumeStager and BlockVolumeUnpublisher only in csi plugin

boylee1111 committed:
Enable snapshottable e2e test for csi pd driver
- add pd driver manifests
- modify snapshottable test case

Commit:
Strip nullable for Server-side apply

lesszyao committed:
add table convertor to componentstatus

soltysh commented on pkg/kubectl/cmd/cmd.go:
```diff
@@ -258,10 +258,19 @@ __kubectl_custom_func() {
             return
             ;;
         kubectl_logs)
+	    __kubectl_parse_get "deploy" "{{ range .items  }}deploy/{{ .metadata.name }} {{ end }}"
```
You'd need to use long names, not aliases. You didn't cover all possible controllers.

Commit:
Update CHANGELOG-1.15.md for v1.15.6.

Commit:
Update CHANGELOG-1.14.md for v1.14.9.

Commit:
Update CHANGELOG-1.16.md for v1.16.3.

MikeSpreitzer committed:
Identify cluster scope by a boolean field rather than a special namespace

MikeSpreitzer committed:
applied gofmt

MikeSpreitzer committed:
updated generated files

k8s-ci-robot committed:
Merge pull request #85018 from yliaog/windep

Windep

k8s-ci-robot committed:
Merge pull request #85075 from hwdef/fix-staticcheck9

pkg/credentialprovider: fix staticcheck warning

k8s-ci-robot committed:
Merge pull request #85084 from barney-s/win-stackdriver-startup-fix

Reorder stackdriver setup in windows startup script

k8s-ci-robot committed:
Merge pull request #85091 from fabriziopandini/alpha-certs-reads-incluster-config

kubeadm: alpha certs command should read in-cluster config

k8s-ci-robot committed:
Merge pull request #85119 from YuikoTakada/remove-unused-CheckNewRSAnnotations

Remove unused function from e2e framework rest.go

k8s-ci-robot committed:
Merge pull request #85137 from rosti/kubeadm-amend-hyperkube-deprecation

kubeadm: Amend the hyperkube deprecation change

k8s-ci-robot committed:
Merge pull request #85150 from ahg-g/ahg-informer

Pass InformerFactory/SharedLister instead of individual informers/listers in scheduler config logic

aramase committed:
add unit tests

alculquicondor committed:
Wait for terminating pods to be deleted

To reduce flakiness of Scheduler Predicates E2E tests

Signed-off-by: Aldo Culquicondor <acondor@google.com>

liggitt committed:
guard kubeadm dependencies on k8s.io/kubernetes

Commit:
Add TODOs for removing invalid e2e dependencies

The e2e core framework and subpackages of e2e framework are defined.
The subpackages can import the core framework, but the core framework
should not import the subpackages. We've defined this dependency rule
after circular depencency issue happened.
This adds TODOs to understand what we should in this rule.

davidz627 committed:
Flip CSIMigration and CSIMigrationGCE to Beta

k8s-ci-robot committed:
Merge pull request #82690 from draveness/feature/graduate-ResourceQuotaScopeSelectors-to-ga

feat: graduate ResourceQuotaScopeSelectors to GA - part2

k8s-ci-robot committed:
Merge pull request #84208 from robscott/endpointslice-e2e

Adding e2e tests covering EndpointSlice and Endpoints Controllers

k8s-ci-robot committed:
Merge pull request #84279 from matthyx/kuberuntime-startupprobe

Add startupProbe result handling to kuberuntime

k8s-ci-robot committed:
Merge pull request #84292 from RainbowMango/pr_add_metrics_flag_to_apiserver

Allow show hidden metrics in kube-apiserver

k8s-ci-robot committed:
Merge pull request #84892 from YuikoTakada/local-metrics-pod

Remove unused func and struct from pod.go

k8s-ci-robot committed:
Merge pull request #85098 from liggitt/admission-config-v1

Promote apiserver.config.k8s.io/v1, kind=AdmissionConfiguration

k8s-ci-robot committed:
Merge pull request #85099 from liggitt/quota-config-v1

Promote apiserver.config.k8s.io/v1, kind=ResourceQuotaConfiguration

k8s-ci-robot committed:
Merge pull request #85108 from rajansandeep/prepcorednsfor1.17-kubeadm

Bump CoreDNS version to 1.6.5 and update manifest

k8s-ci-robot committed:
Merge pull request #85109 from rajansandeep/prepcorednsfor1.17-kube-up

Bumps CoreDNS to 1.6.5 and updates manifest for kube-up

k8s-ci-robot committed:
Merge pull request #85138 from liggitt/webhook-config-v1

Promote apiserver.config.k8s.io/v1, kind=WebhookAdmissionConfiguration

k8s-ci-robot committed:
Merge pull request #85151 from hypnoglow/scheduler-default-algo-source

scheduler: make algorithm source an option

k8s-ci-robot committed:
Merge pull request #85156 from chuckha/remove-infinte-poll

[kubeadm] Remove nested retries

k8s-ci-robot committed:
Merge pull request #85157 from alculquicondor/refactor/selector

Store topology spread constraints in metadata with labels.Selector

k8s-ci-robot committed:
Merge pull request #85158 from andyzhangx/vmss-dirty-cache

fix vmss dirty cache issue

k8s-ci-robot committed:
Merge pull request #85180 from k-toyoda-pi/use_log_e2e_storage_utils

Use log functions of core framework on e2e/storage/utils

liu-cong committed:
Update bucket for scheduler framework latency histograms.

Commit:
Flip CSIMigrationAWS flag to be beta and off by default

k8s-ci-robot committed:
Merge pull request #85004 from deads2k/dynamic-agg-cert

dynamic reload cluster authentication info for aggregated API servers

k8s-ci-robot committed:
Merge pull request #85152 from mikedanese/tokbench

report cache misses in cached token authenticator benchmark

k8s-ci-robot committed:
Merge pull request #85194 from hwdef/fix-link

fix wrong link

k8s-ci-robot committed:
Merge pull request #85202 from RainbowMango/pr_add_Reset_API_to_component-base

Add Reset() API to stability framework

k8s-ci-robot committed:
Merge pull request #85217 from kaoskater08/kaoskater08/fix-golint-pkg-apis-extensions

fixed golint errors in pkg/apis/extensions register.go and types.go

k8s-ci-robot committed:
Merge pull request #85219 from mm4tt/lease_cache_size

Increase cache size for leases

k8s-ci-robot committed:
Merge pull request #85220 from liggitt/revert-licenses

Revert #76586, restructure LICENSES file generation

robscott committed:
Updating kube-proxy to support new EndpointSlice address types

This includes IPv4 and IPv6 address types and IPVS dual stack support.
Importantly this ensures that EndpointSlices with a FQDN address type
are not processed by kube-proxy.

davidz627 committed:
Update attachdetach-controller role to include permissions to get, list, and watch csinodes for CSIMigration

Commit:
Move test inputs for EncryptionConfiguration tests into testdata.

prameshj committed:
Specify a port range to ILB firewall rule create.

k8s-ci-robot committed:
Merge pull request #83057 from bclau/windows/containerd

Windows: Fixes termination-file mounting support for containerd

k8s-ci-robot committed:
Merge pull request #84390 from robscott/endpointslice-beta

Promoting EndpointSlices to beta

k8s-ci-robot committed:
Merge pull request #84771 from MikeSpreitzer/refactor-priority-config

Refactored PriorityLevelConfiguration

k8s-ci-robot committed:
Merge pull request #85023 from MrHohn/svc-lb-ga

Promote service load balancer finalizer to GA

k8s-ci-robot committed:
Merge pull request #85153 from codenrhoden/mount-no-exec-int

Retire mount.Exec for k8s.io/utils/exec

k8s-ci-robot committed:
Merge pull request #85226 from alculquicondor/flake/stable_cluster

Wait for terminating pods to be deleted

k8s-ci-robot committed:
Merge pull request #85229 from liggitt/kubeadm-imports

guard kubeadm dependencies on k8s.io/kubernetes

k8s-ci-robot committed:
Merge pull request #85236 from liu-cong/metrics-bucket

Update bucket for scheduler framework latency histograms.

Commit:
Add logic for translating zone/zones/allowtopologies

k8s-ci-robot committed:
Merge pull request #85010 from leakingtapan/ebs-migration

Add translation logic for EBS storage class fstype parameter

k8s-ci-robot committed:
Merge pull request #85129 from timyinshi/master

modify istio url:https://istio.io/news/2018/announcing-0.8/

k8s-ci-robot committed:
Merge pull request #85149 from oomichi/move-util-4

Move functions from e2e/framework/util.go Part-4

k8s-ci-robot committed:
Merge pull request #85192 from MikeSpreitzer/fq-impl

Added fair queuing for server requests

k8s-ci-robot committed:
Merge pull request #85227 from apelisse/update-smd

Update structured-merge-diff to latest version

yutedz committed:
Correct the checking of robinIndex

MikeSpreitzer committed:
Brushed up fairqueuing package

This commit responds to the comments on PR #85192 that were not yet
addressed at the time it merged, apart from the one fixed in PR

Generalized fairqueuing to allow for zero queues, to support a
priority level that limits concurrency but does no queuing.

liggitt committed:
Include *intstr.IntOrString in API compatibility tests

k8s-ci-robot committed:
Merge pull request #82031 from codenrhoden/mount-comments

Additional mount comments

k8s-ci-robot committed:
Merge pull request #82878 from adelina-t/fix_82876

Check for terminated reason appropriate for containerD and dockershim

k8s-ci-robot committed:
Merge pull request #83394 from bertinatto/convert_pv_no_affinity

Convert existing PVs to use volume topology in VolumeBinderPredicate

k8s-ci-robot committed:
Merge pull request #83839 from RainbowMango/pr_hide_proxy_deprecated_metrics

Turn off proxy metrics that have been deprecated

k8s-ci-robot committed:
Merge pull request #84472 from gab-satchi/windows-build-label

Adds Windows build information as a label on the node

k8s-ci-robot committed:
Merge pull request #84832 from gnufied/update-csi-version

bump CSI version to 1.2.0

MikeSpreitzer committed:
Regenerated pb.go for flowcontrol/v1alpha1

MikeSpreitzer committed:
Renamed FeatureGate RequestManagement to APIPriorityAndFairness

The old name is too broad, we wanted a name that is more specific to
the actual feature.

This is an alpha gate, and no release has yet associated any
functionality with this gate.

k8s-ci-robot committed:
Merge pull request #80572 from knight42/fix/scale-cr

Fix missing resource version when updating the scale subresource of custom resource

k8s-ci-robot committed:
Merge pull request #85115 from aramase/azure-disk-lock

azure: remove disk locks per vm during attach/detach

fabriziopandini committed:
add retry to etcd operations

k8s-ci-robot committed:
Merge pull request #84958 from kkmsft/disk_fixes

Azure : filter disks with ToBeDetached flag

k8s-ci-robot committed:
Merge pull request #85258 from liggitt/fuzz-pointer-intstr

Include *intstr.IntOrString in API compatibility test fixtures

mm4tt committed:
pager.go: Don't set ResourveVersion on subsequent List calls

Ref. https://github.com/kubernetes/kubernetes/issues/85221#issuecomment-553748143

k8s-ci-robot committed:
Merge pull request #85077 from gongguan/DeleteOptions

add DeleteOptions conversion

liggitt committed:
Fix --resource-version handling in kubectl

danwinship committed:
Update debian-iptables image digests for v12.0.1 build

k8s-ci-robot committed:
Merge pull request #85201 from fabriziopandini/add-retry-to-etcd

kubeadm: add retry to etcd calls

yutedz committed:
Check error return from closing connection

mkimuram committed:
Change mount.NewOSExec to utilexec.New

ahg-g committed:
rename PluginConfigArgs to AlgorithmConfigArgs

k8s-ci-robot committed:
Merge pull request #84046 from bclau/tests/agnhost-guestbook-replacement

Tests/agnhost guestbook replacement

k8s-ci-robot committed:
Merge pull request #84335 from mrkm4ntr/fix-node-evaluation

Add unit test to catch scheduler's node order evaluation regressions

k8s-ci-robot committed:
Merge pull request #84859 from denkensk/move-volumebind-behind-permit

Move pod bindVolumes behind RunPermitPlugins

liggitt committed:
Use CSINodes v1 API in scheduler

robscott committed:
Ensuring EndpointSlice controller does not start when feature gate or
API are disabled

boylee1111 committed:
update pd csi driver controller manifests

k8s-ci-robot committed:
Merge pull request #84732 from khenidak/fix-disable-dualstack

Fix a CM panic when ipam tries to lock an out of range pre existing cidr

k8s-ci-robot committed:
Merge pull request #85162 from apelisse/strip-nullable

Strip nullable for Server-side apply

k8s-ci-robot committed:
Merge pull request #85174 from zhouya0/add_table_convertor_componentstatus

add table convertor to componentstatus

k8s-ci-robot committed:
Merge pull request #85294 from ahg-g/ahg-15k

Rename PluginConfigArgs to AlgorithmConfigArgs

codenrhoden committed:
Remove pkg/util/mount (moved out of tree)

This patch removes pkg/util/mount completely, and replaces it with the
mount package now located at k8s.io/utils/mount. The code found at
k8s.io/utils/mount was moved there from pkg/util/mount, so the code is
identical, just no longer in-tree to k/k.

davidz627 committed:
De-race some CSI unit tests that were initializing the plugin manager…ger (and plugins) twice. Set some const variables earlier to support node info manager initialization and wait for initialization to complete before finishing plugin setup.

liggitt committed:
Avoid constructing table printer on every componentstatus request

k8s-ci-robot committed:
Merge pull request #83964 from Jefftree/bdd-conformance

Initial Implementation for kubetestgen for Conformance.

k8s-ci-robot committed:
Merge pull request #84227 from soltysh/fix_scale_doc

Drop job from scale description

k8s-ci-robot committed:
Merge pull request #84439 from danielqsj/f-apparmor

refactor apparmor utils in e2e

k8s-ci-robot committed:
Merge pull request #84660 from mkimuram/refactor-block-lock

Refactor block volume's descriptor lock logic

k8s-ci-robot committed:
Merge pull request #84674 from hwdef/fix-staticcheck4

pkg/client: fix staticcheck warning

k8s-ci-robot committed:
Merge pull request #84752 from seans3/json_yaml_printers

Move json,yaml,jsonpath printers to correct locations

k8s-ci-robot committed:
Merge pull request #85117 from hwdef/fix-staticcheck10

pkg/kubeapiserver: fix staticcheck warning

k8s-ci-robot committed:
Merge pull request #85175 from liggitt/golang-org-comments

Add comments to explain golang.org replace directives

k8s-ci-robot committed:
Merge pull request #85230 from oomichi/add-todo-issue81245

Add TODOs for removing invalid e2e dependencies

k8s-ci-robot committed:
Merge pull request #85246 from robscott/endpointslice-dualstack-proxy

Updating kube-proxy to support new EndpointSlice address types

davidz627 committed:
Panic on error when creating fake volume host, add a TODO to clean up later

k8s-ci-robot committed:
Merge pull request #83058 from bclau/tests/windows-tests-support

tests: Fixes tests for Windows (containerd, RunAsUserName)

k8s-ci-robot committed:
Merge pull request #83491 from dcbw/winship-iptables-owner

pkg/util/iptables: add Dan Winship to approvers

k8s-ci-robot committed:
Merge pull request #83680 from bclau/tests/network-large-requests

tests: Adds large requests tests

k8s-ci-robot committed:
Merge pull request #84304 from liggitt/all-beta

Add support for --runtime-config=api/beta=false, --feature-gates=AllBeta=false

k8s-ci-robot committed:
Merge pull request #84900 from MikeSpreitzer/add-namespace-to-rule

Enable Priority and Fairness to discriminate on target namespace

k8s-ci-robot committed:
Merge pull request #85147 from yutedz/devmgr-rm-contents

Continue removing file in ManagerImpl#removeContents

k8s-ci-robot committed:
Merge pull request #85252 from prameshj/fwrules-port

Specify a port range to ILB firewall rule create.

k8s-ci-robot committed:
Merge pull request #85257 from yutedz/queueset-robin-idx

Correct the checking of robinIndex

k8s-ci-robot committed:
Merge pull request #85285 from liggitt/kubectl-resource-version

Fix --resource-version handling in kubectl

k8s-ci-robot committed:
Merge pull request #85301 from robscott/endpointslice-beta-fix

Ensuring EndpointSlice controller does not start when feature gate or API are disabled

k8s-ci-robot committed:
Merge pull request #83505 from misterikkit/govmomi-dep

Update dependency vmware/govmomi to v0.20.3

k8s-ci-robot committed:
Merge pull request #84295 from aojea/iptableslogs

Improve iptables logging

k8s-ci-robot committed:
Merge pull request #84424 from mikedanese/expcache

Add an expiring cache for the caching token authenticator

k8s-ci-robot committed:
Merge pull request #84747 from mkimuram/refactor-mapper

Refactor mapper/unmapper of block volume

k8s-ci-robot committed:
Merge pull request #85014 from dekkagaijin/master

let standalone npd use kubelet credentials

k8s-ci-robot committed:
Merge pull request #85169 from boylee1111/enable_snapshot_e2e_test_for_pd

Enable snapshot e2e test for csi pd driver

k8s-ci-robot committed:
Merge pull request #85249 from immutableT/testdata-for-encryption-config

Move test inputs for EncryptionConfiguration tests into testdata.

k8s-ci-robot committed:
Merge pull request #85260 from MikeSpreitzer/reqmgmt-rename

Renamed FeatureGate RequestManagement to APIPriorityAndFairness

k8s-ci-robot committed:
Merge pull request #85298 from liggitt/scheduler-csi-v1

Use CSINodes v1 API in scheduler

k8s-ci-robot committed:
Merge pull request #85309 from liggitt/cs-table-handler

Avoid constructing table printer on every componentstatus request

imroc committed:
kube-proxy: sync rules when current node labels change detected

davidz627 committed:
Revert "Enable snapshot e2e test for csi pd driver"

k8s-ci-robot committed:
Merge pull request #74026 from mkimuram/issue/73773

Separate staging/publish and unstaging/unpublish logics for block

k8s-ci-robot committed:
Merge pull request #83098 from ddebroy/disable-intree

CSI Migration phase 2: disable probing of in-tree plugins

k8s-ci-robot committed:
Merge pull request #83576 from andrewsykim/udpate-netlink

update github.com/vishvananda/netlink to v1.0.0

k8s-ci-robot committed:
Merge pull request #83896 from mars1024/modify/cni_log

modify error output in cniNetworkPlugin

tallclair committed:
Restrict mirror pod owner references (#84657)

* Restrict mirror pod owners.

See http://git.k8s.io/enhancements/keps/sig-auth/20190916-noderestriction-pods.md

* Address feedback, refactor test

* Verify node owner UID

k8s-ci-robot committed:
Merge pull request #83123 from aramase/dualstack-downward-api

Dualstack downward api

k8s-ci-robot committed:
Merge pull request #79993 from aramase/controller-manager-multiple-cidr

Allow multiple node cidr masks in kube-controller-manager

davidz627 committed:
Add CSINodes to AttachDetachControllerRecovery test

davidz627 committed:
Jump out of spec translation early if the spec is not migratable. Unit tests work after all!

k8s-ci-robot committed:
Merge pull request #84875 from oomichi/move-test_context

Move RegisterNodeFlags() to e2e_node test

k8s-ci-robot committed:
Merge pull request #85259 from MikeSpreitzer/fq-followup

Brushed up fairqueuing package

k8s-ci-robot committed:
Merge pull request #85322 from davidz627/revert-85169-enable_snapshot_e2e_test_for_pd

Revert "Enable snapshot e2e test for csi pd driver"

mm4tt committed:
Increase cache size for endpointslices.

It should match what we have for endpoints as endpointslices is
replacing them and # updates of endpointslices should be roughly the
same as # updates of endpoints.

Otherwise, kube-proxy may start loosing watches for endpointslices (we believe this is
what currently happens in scale tests, will create an issue for that
soon).

k8s-ci-robot committed:
Merge pull request #72046 from m1093782566/service-topology-api

Service Topology implementation

k8s-ci-robot committed:
Merge pull request #84754 from davidz627/fix/uniqueName

Update inline volume translated PV Name to be unique per disk so that staging paths are unique

jkaniuk committed:
Addon Manager: Fix parsing KUBECTL_EXTRA_PRUNE_WHITELIST

When there are two or more values in KUBECTL_EXTRA_PRUNE_WHITELIST
kube-addons.sh wrongly generates "--prune-whitelist value1 value2"
Which kubectl parses as additional unknown flag[s]

Change-Id: I7661bda41ab7435e0e65eb0a662d1b16d2c14991

k8s-ci-robot committed:
Merge pull request #85272 from mm4tt/pager_fix

pager.go: don't set ResourceVersion on subsequent List calls

andrewsykim committed:
service topology: remove redundant API validation for local external traffic policy

Signed-off-by: Andrew Sy Kim <kiman@vmware.com>

ereslibre committed:
kubeadm: use the right name for the no validator function

rosti committed:
kubeadm: Use only stdout when calling kubelet for its version

Currently this uses the combined kubelet output (stdout + stderr), but this
causes parsing issues if the kubelet logs something on stderr.
Thus we ignore the entire stderr and use stdout only.

We do disable a couple of tests here. That is because the fakeexecer only
supports combined output and return a "not supported" error if `.Output()`
gets invoked thus permanently failing those.

Signed-off-by: Rostislav M. Georgiev <rostislavg@vmware.com>

k8s-ci-robot committed:
Merge pull request #83838 from RainbowMango/pr_hide_scheduler_deprecated_metrics

Turn off scheduler metrics that have been deprecated

sttts committed:
apiextensions: factor out predicate iterator in validation

sttts committed:
apiextensions: ratcheting update validation for atomic item of set list-type

k8s-ci-robot committed:
Merge pull request #85284 from yutedz/svr-check-close-ret

Check error return from closing connection

k8s-ci-robot committed:
Merge pull request #85305 from codenrhoden/remove-mount-pkg

Remove pkg/util/mount (moved out of tree)

k8s-ci-robot committed:
Merge pull request #85339 from mm4tt/endpointslice_cache

Increase cache size for EndpointSlices.

k8s-ci-robot committed:
Merge pull request #85346 from andrewsykim/service-topology-validtaion

service topology: remove redundant API validation for local external traffic policy

robscott committed:
Reverting managed-by-setup annotation

This ended up causing far more problems than it was worth, especially
given that it just attempted to provide backwards compatibility with
the alpha release.

liggitt committed:
Fix label mutation in endpoints controller

k8s-ci-robot committed:
Merge pull request #85359 from robscott/endpointslice-label-fix

Reverting managed-by-setup annotation

robscott committed:
Disabling EndpointSlice feature gate by default

Given the significance this change would have we've decided to hold off
on enabling this by default until we can have better test coverage and
more real world usage of the feature.

aramase committed:
add comments for compatibility

k8s-ci-robot committed:
Merge pull request #85348 from ereslibre/validator-names

kubeadm: use the right name for the no validator function

k8s-ci-robot committed:
Merge pull request #85361 from liggitt/endpoint-label

Fix label mutation in endpoints controller

mikedanese committed:
inline GC in expiring cache

This allows us to drop the background goroutine with negligable
difference in performance.

mikedanese committed:
fix nasty bug

This needs to be the opposite check of what we do in Get(). Introduced
in a fixup commit.

liggitt committed:
Add single-item list/watch to delegated authentication reader role

k8s-ci-robot committed:
Merge pull request #85231 from davidz627/flags

Flip CSIMigration and CSIMigrationGCE to Beta

k8s-ci-robot committed:
Merge pull request #85365 from robscott/endpointslice-default-off

Disabling EndpointSlice feature gate by default

liggitt committed:
Skip Node printing in kubectl column test

mikedanese committed:
fix stress test: it's not doing anything

k8s-ci-robot committed:
Merge pull request #85351 from rosti/kubeadm-fix-kubelet-version-check

kubeadm: Use only stdout when calling kubelet for its version

k8s-ci-robot committed:
Merge pull request #85367 from aramase/node-cidr-followup

[kube controller manager] add comments for compatibility

k8s-ci-robot committed:
Merge pull request #85375 from liggitt/delegated-list-watch

Add single-item list/watch to delegated authentication reader role

k8s-ci-robot committed:
Merge pull request #85377 from liggitt/node-kubectl

Skip Node printing in kubectl column test

k8s-ci-robot committed:
Merge pull request #84722 from sttts/sttts-crd-list-map-type-atomic-validations

apiextensions: validate list-type and map-type "atomic"

k8s-ci-robot committed:
Merge pull request #85370 from mikedanese/expcache

fix terrible bug and inline GC for expiring cache

k8s-ci-robot committed:
Merge pull request #85251 from leakingtapan/ebs-migraiton-zone

Add CSI migration logic for EBS storageclass zone/zones/topology

sttts committed:
apiextensions: freeze & doc supported v1.16 OpenAPI formats

k8s-ci-robot committed:
Merge pull request #80854 from aojea/hostportv6

Add IPv6 support to kubenet hostport

k8s-ci-robot committed:
Merge pull request #84861 from sttts/sttts-crd-defaulting-conformance

e2e: promote CRD defaulting test to conformance test

k8s-ci-robot committed:
Merge pull request #82966 from danwinship/iptables-nft

update images to Debian buster, detect iptables mode

k8s-ci-robot committed:
Merge pull request #85386 from mikedanese/fixstresstest

cache.Expiring: fix stress test: it's not doing anything

k8s-ci-robot committed:
Merge pull request #85381 from sttts/sttts-apiextensions-fix-openapi-formats

apiextensions: freeze and document supported v1.16 OpenAPI formats

k8s-ci-robot committed:
Merge pull request #83837 from RainbowMango/pr_hide_apiserver_deprecated_metrics

Turn off apiserver deprecated metrics

k8s-ci-robot committed:
Merge pull request #85344 from jkaniuk/prune-whitelist

Addon Manager: Fix parsing KUBECTL_EXTRA_PRUNE_WHITELIST

Commit:
Update CHANGELOG-1.17.md for v1.17.0-beta.2.

k8s-ci-robot committed:
Merge pull request #85237 from leakingtapan/ebs-migration-flag

Flip CSIMigrationAWS flag to be beta and off by default

liggitt committed:
Plumb authorization webhook version from CLI to config

k8s-ci-robot committed:
Merge pull request #85441 from liggitt/authorizer-version-plumbing

Plumb authorization webhook version from CLI to config

sroskelley committed:
use long names

Commit:
use long names

Commit:
use long names

sroskelley committed:
Merge branch 'master' of github.com:sroskelley/kubernetes

sroskelley committed:
fix spaces

sroskelley committed:
fix spaces

sroskelley commented on pkg/kubectl/cmd/cmd.go:
```diff
@@ -258,7 +258,7 @@ __kubectl_custom_func() {
+            __kubectl_parse_get "deployments" "{{ range .items  }}deploy/{{ .metadata.name }} {{ end }}"
             __kubectl_parse_get "statefulsets" "{{ range .items  }}sts/{{ .metadata.name }} {{ end }}"
             __kubectl_parse_get "replicasets" "{{ range .items  }}rs/{{ .metadata.name }} {{ end }}"
             __kubectl_parse_get "jobs" "{{ range .items  }}job/{{ .metadata.name }} {{ end }}"
```
Use long controller names per @soltysh request. But I'm not aware of any other controllers that have support for fetching logs.

sroskelley committed:
Merge branch 'master' of github.com:sroskelley/kubernetes
